src/transformers/models/marian/convert_marian_to_pytorch.py

```diff
@@ -622,6 +622,12 @@ def load_marian_model(self) -> MarianMTModel:
             bias_tensor = nn.Parameter(torch.FloatTensor(self.final_bias))
             model.model.decoder.embed_tokens.weight = decoder_wemb_tensor
 
+        # handle tied embeddings, otherwise "from_pretrained" loads them incorrectly
+        if self.cfg["tied-embeddings"]:
+            wemb_tensor = nn.Parameter(torch.FloatTensor(self.wemb))
+            bias_tensor = nn.Parameter(torch.FloatTensor(self.final_bias))
```

Comment:
I don't see where these are used?

Reply:
oh, yeah, makes sense. They are not used anymore

Reply:
Weird I cannot reply to the above comment. Anyway, I tested FSDP with `fsdp_cpu_ram_efficient_loading: true`, and looks like tying weights do not have big effect on CPU memory. But to be clear, summoning @pacman100 to confirm that as he was the PR contributor